src/net_processing.cpp

```diff
@@ -4788,6 +4911,9 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
                     if (ping_time.count() >= 0) {
                         // Let connman know about this successful ping-pong
                         pfrom.PongReceived(ping_time);
+                        if (pfrom.IsSensitiveRelayConn()) {
+                            pfrom.fDisconnect = true;
```

Comment:
(I just realized I answered my own question from https://github.com/bitcoin/bitcoin/pull/27509#discussion_r1214675802)

But still, don't we also send automatic PINGs ? Could that result in premature disconnection?

Reply:
> Why disconnect the sensitive connection on PONG?

Because then we assume the TX message that preceded the PING message has been processed.

> Could that ever happen before we send our tx messages out?

Yes, if the peer sends us unsolicited PONG without us sending PING before that.

> But still, don't we also send automatic PINGs?

Yes, and for the sensitive relay connections this should be the first PING. The point is to send TX before any PING.

Reply:
Oh, yes, this is in the "nonce match" branch of the logic that handles PONG. So, the answer to "Could that ever happen before we send our tx messages out?" is just "No".